spec.emu

```diff
@@ -1,23 +1,336 @@
+<emu-clause id="durationformat-objects">
+  <h1>DurationFormat Objects</h1>
 
-<emu-clause id="sec-demo-clause">
-  <h1>This is an emu-clause</h1>
-  <p>This is an algorithm:</p>
-  <emu-alg>
-    1. Let _proposal_ be *undefined*.
-    1. If IsAccepted(_proposal_),
-      1. Let _stage_ be *0*.
-    1. Else,
-      1. Let _stage_ be *-1*.
-    1. Return ? ToString(_proposal_).
-  </emu-alg>
+  <emu-clause id="sec-intl-durationformat-abstracts">
+    <h1>Abstract Operations for DurationFormat Objects</h1>
+
+    <p>Several DurationFormat algorithms use values from the following table, which provides internal slots and property names for Duration instances, and which is also used to key Records of templates to format individual Duration components: </p>
+
+    <emu-table id="table-duration-components">
+      <emu-caption>Components of Duration Instances</emu-caption>
+      <table class="real-table">
+        <thead>
+          <tr>
+            <th>Internal Slot</th>
+            <th>Property</th>
+          </tr>
+        </thead>
+        <tr>
+          <td>[[Years]]</td>
+          <td>`"year"`</td>
+        </tr>
+        <tr>
+          <td>[[Months]]</td>
+          <td>`"months"`</td>
+        </tr>
+        <tr>
+          <td>[[Weeks]]</td>
+          <td>`"weeks"`</td>
+        </tr>
+        <tr>
+          <td>[[Days]]</td>
+          <td>`"days"`</td>
+        </tr>
+        <tr>
+          <td>[[Hours]]</td>
+          <td>`"hours"`</td>
+        </tr>
+        <tr>
+          <td>[[Minutes]]</td>
+          <td>`"minutes"`</td>
+        </tr>
+        <tr>
+          <td>[[Seconds]]</td>
+          <td>`"seconds"`</td>
+        </tr>
+        <tr>
+          <td>[[Milliseconds]]</td>
+          <td>`"milliseconds"`</td>
+        </tr>
+        <tr>
+          <td>[[Microseconds]]</td>
+          <td>`"microseconds"`</td>
+        </tr>
+        <tr>
+          <td>[[Nanoseconds]]</td>
+          <td>`"nanonseconds"`</td>
+        </tr>
+      </table>
+    </emu-table>
+
+    <emu-note>The [[Weeks]] field is still under discussion in Temporal. For more, check out the <a href="https://github.com/tc39/proposal-temporal/issues/532">Github thread</a> on the Temporal issue tracker.</emu-note>
+
+    <emu-clause id="sec-partitiondurationformatpattern" aoid="PartitionDurationFormatPattern">
+      <h1>PartitionDurationFormatPattern ( _durationFormat_, _duration_ )</h1>
+
+      <p>The PartitionDurationFormatPattern abstract operation is called with the arguments _durationFormat_ (which must be an object initialized as a DurationFormat) and _duration_ (which must be a Record), and creates the corresponding parts according to the effective locale and the formatting options of _durationFormat_. The following steps are taken:</p>
+
+      <emu-alg>
+        1. Let _result_ be a new empty List.
+        1. If _durationFormat_.[[Style]] is `"dotted"`, then
+          1. Let _patternParts_ be PartitionPattern(_durationFormat_.[[Templates]]).
```

Comment:
I guess I'm having trouble finding where this becomes a pattern... isn't this always supposed to be a Record with the duration-like fields?

Reply:
It becomes a pattern for `style: "dotted"` and is otherwise a Record.

Reply:
I added more description for it down below.